nilearn/datasets/utils.py

```diff
@@ -506,7 +525,13 @@ def _fetch_file(url, data_dir, resume=True, overwrite=False,
     If, for any reason, the download procedure fails, all downloaded files are
     removed.
     """
-    handlers = handlers if handlers else []
+    if session is None:
+        with requests.Session() as session:
+            session.mount("ftp:", _NaiveFTPAdapter())
```

Comment:
I'm a bit confused why we'd default to ftp if session is None. Maybe a docstring could be useful, here !

Reply:
it is not a default, we just add a transport adapter for ftp because the requests sessions only provide one adapter that only handles http and https -- [more details here](https://requests.readthedocs.io/en/master/user/advanced/#transport-adapters)